src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/ProfileEndpoints.cs

```diff
@@ -34,18 +35,38 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
 /// <summary>
 /// User profile information
 /// </summary>
-public record UserProfile(
-    string? FirstName,
-    string? LastName,
-    string? MiddleName,
-    string? PreferredName,
-    string? AlternateContactPhone,
-    DateOnly? DateOfBirth,
-    string Email,
-    string Phone,
-    Address? ResidentialAddress,
-    Address? MailingAddress
-    );
+public record UserProfile {
+  public string? FirstName { get; set; }
+  public string? LastName { get; set; }
+  public string? MiddleName { get; set; }
+  public string? PreferredName { get; set; }
+  public string? AlternateContactPhone { get; set; }
+  public DateOnly? DateOfBirth { get; set; }
+  public string Email { get; set; } = null!;
+  public string Phone { get; set; } = null!;
+  public Address? ResidentialAddress { get; set; }
+  public Address? MailingAddress { get; set; }
+  public IEnumerable<PreviousName> PreviousNames { get; set; } = Array.Empty<PreviousName>();
+};
+
+/// <summary>
+/// Previous Name
+/// </summary>
+public record PreviousName ([Required] string? FirstName, [Required] string? LastName)
```

Comment:
if these fields are required why do we have them nullable?

Reply:
Will clean up in next previous name branch